cli/provisionerjobs.go

```diff
@@ -0,0 +1,120 @@
+package cli
+
+import (
+	"fmt"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/cli/cliui"
+	"github.com/coder/coder/v2/coderd/util/slice"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/serpent"
+)
+
+func (r *RootCmd) provisionerJobs() *serpent.Command {
+	cmd := &serpent.Command{
+		Use:   "jobs",
+		Short: "View and manage provisioner jobs.",
+		Handler: func(inv *serpent.Invocation) error {
+			return inv.Command.HelpHandler(inv)
+		},
+		Aliases: []string{"job"},
+		Children: []*serpent.Command{
+			r.provisionerJobsList(),
+		},
+	}
+	return cmd
+}
+
+func (r *RootCmd) provisionerJobsList() *serpent.Command {
+	type provisionerJobRow struct {
+		codersdk.ProvisionerJob `table:"provisioner_job,recursive_inline"`
+		OrganizationName        string `json:"organization_name" table:"organization"`
+		Queue                   string `json:"-" table:"queue"`
+	}
+
+	var (
+		client     = new(codersdk.Client)
+		orgContext = NewOrganizationContext()
+		formatter  = cliui.NewOutputFormatter(
+			cliui.TableFormat([]provisionerJobRow{}, []string{"created at", "id", "organization", "status", "type", "queue", "tags"}),
+			cliui.JSONFormat(),
+		)
+		status []string
+		limit  int64
+	)
+
+	cmd := &serpent.Command{
+		Use:     "list",
+		Short:   "List provisioner jobs",
+		Aliases: []string{"ls"},
+		Middleware: serpent.Chain(
+			serpent.RequireNArgs(0),
+			r.InitClient(client),
+		),
+		Handler: func(inv *serpent.Invocation) error {
+			ctx := inv.Context()
+			org, err := orgContext.Selected(inv, client)
+			if err != nil {
+				return xerrors.Errorf("current organization: %w", err)
+			}
+
+			jobs, err := client.OrganizationProvisionerJobs(ctx, org.ID, &codersdk.OrganizationProvisionerJobsOptions{
+				Status: slice.ToStringEnums[codersdk.ProvisionerJobStatus](status),
+				Limit:  int(limit),
```

Comment:
nit: paging?

Reply:
I want to keep it simple for now, I realized we don't have a precedent on the CLI for paging so I didn't want to introduce it here. Granted `--limit` is also new, so I could remove that too but I was worried about how many entries there will be.

The original issue #15084 mentioned we should have "created at" (or rather "created after"), and I added that initially, but then I realized we don't have a precedent for filtering on time on the CLI either. (There's even a lack of serpent support there.) For this reason both pagination and created at filter have been omitted and should IMO be part of a grander plan for how to support these in a unified way on the CLI.